uwblueprint/sistering
# Create Posting CRUD Methods and GraphQL Resolvers

## Ticket link

Closes #50 



## Implementation description
- Implement CRUD methods:
  - getPosting
  - getPostings
  - createPosting
  - updatePosting
  - deletePosting
- Create corresponding GraphQL resolvers and DTOs
- Modified Prerequisites Model in schema as requested in ticket and generated associated prisma migration file 


## Steps to test
1. Start docker and run `docker-compose up --build`
2. We need row(s) in the skills, branches, and employees tables for testing and will need to create those with raw SQL for now. Run the following commands:
```bash
docker exec -it sistering_db /bin/bash
psql -U postgres -d sistering
INSERT INTO branches (name) VALUES ('Branch1');
INSERT INTO skills (name) VALUES ('Skill1');
INSERT INTO employees (userId, branch_id) VALUES ('1','1');
```
3. Run `docker exec -it sistering_backend /bin/bash -c "npx prisma generate"` and run the lastest migration files using `npx prisma migrate deploy`
4. Go to http://localhost:5000/graphql and verify that all GraphQL queries and mutations introduced in this ticket work as expected


## What should reviewers focus on?
* Verify that GraphQL endpoints work as expected


## Checklist
